### PR TITLE
Feature/49901 Migrate product page

### PR DESCRIPTION
**Pull request by Aekal**
dev-49901

### CODE REVIEW

File: components/02-elements/swatch/mocks/icon-swatch.json [hunk 1]
```diff
@@ -7,28 +7,28 @@
...
-      "aria-label" : "Red",
+      "ariaLabel" : "Red",
       "text": "L",
       "value": "L"
     },
     {
       "id": "opt-color-white-4",
       "style": "background-image: url('../../images/swatch/swatch-white.jpg')",
-      "aria-label": "White",
+      "ariaLabel": "White",
```
Comment (szafran89) [inline components/02-elements/swatch/mocks/icon-swatch.json]: We should remove ariaLabel from mock, IMO we don't get it from backend

File: components/03-modules/additional-content/AdditionalContent.stories.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      picture
+    }
+  },
+  template: `
+    <app>
+      <alpaca-additional-content
+        :picture="picture"
+        pictureAlt="Sample alt text"
+        firstColumnText="First column"
+        secondColumnText="Second column"
```
Comment (szafran89) [inline components/03-modules/additional-content/AdditionalContent.stories.js]: https://vuejs.github.io/eslint-plugin-vue/rules/attribute-hyphenation.html#rule-details for all above props

File: components/03-modules/additional-content/AdditionalContent.vue [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      default: ''
+    }
+  }
+}
+</script>
+
+
+<style lang="scss">
+$additional-content__image-margin       : 0 0 41px !default;
+$additional-content__image-margin\@large: 0 0 32px !default;
```
Comment (szafran89) [inline components/03-modules/additional-content/AdditionalContent.vue]: can we use `$spacer`

File: components/04-views/product-view/ProductView.html [hunk 4]
```diff
@@ -0,0 +1,211 @@
+<alpaca-container>
+  <div class="product-view">
+    <alpaca-breadcrumbs
+      :breadcrumbs="breadcrumbs"
+    />
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: set one line

File: components/04-views/product-view/ProductView.html [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    <div class="product-view__wrapper">
+      <div class="product-view__gallery">
+        <alpaca-gallery
+          :images="gallery.images"
+        />
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: set one line

File: components/04-views/product-view/ProductView.html [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      </div>
+      <div class="product-view__main-details">
+        <div class="product-view__heading">
+          <alpaca-heading
+            tag="h1"
+            level="1"
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: do we need tag and level, Is is not the same result?

File: components/04-views/product-view/ProductView.html [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+            v-else-if="product.qty > 0"
+            class="product-view__stock-status"
+          >
+            In stock
+          </span>
+          <span
+            v-else
+            class="product-view__stock-status product-view__stock-status--out-of-stock"
+          >
+            Out of stock
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: we need move outside all strings

File: components/04-views/product-view/ProductView.html [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+            v-if="product.qty > 0 && product.qty <= 5"
+            class="product-view__stock-status product-view__stock-status--alert"
+          >
+            In stock - only {{ product.qty }} left
+          </span>
+          <span
+            v-else-if="product.qty > 0"
+            class="product-view__stock-status"
+          >
+            In stock
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: we need move outside all strings

File: components/04-views/product-view/ProductView.html [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+            large
+            :old-price="product.price.old"
+            :special-price="product.price.special"
+          />
+          <!-- TODO: Replace with new stock component -->
+          <span
+            v-if="product.qty > 0 && product.qty <= 5"
+            class="product-view__stock-status product-view__stock-status--alert"
+          >
+            In stock - only {{ product.qty }} left
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: we need move outside all strings

File: components/04-views/product-view/ProductView.html [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+        <div class="product-view__main-actions">
+          <div class="product-view__qty">
+            <!-- TODO: Add method for price update -->
+            <alpaca-quantity-update/>
+          </div>
+          <div class="product-view__action-elements">
+            <div class="product-view__add-to-cart">
+              <!-- TODO: Add methods for action buttons -->
+              <alpaca-button class="product-view__add-to-cart-btn">
+                Add to cart
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: we need move outside all strings

File: components/04-views/product-view/ProductView.html [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+      </alpaca-heading>
+      <alpaca-slider>
+        <slide
+          class="slider__item product-view__slide-item"
+          v-for="(product, index) in product.featuredProducts"
+          :key="index"
+        >
+          <alpaca-catalog-grid-item
+            :product="product"
+          />
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.html]: please keep it in one line

File: components/04-views/product-view/ProductView.stories.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+          link="#"
+        />
+        <alpaca-product-view
+          :breadcrumbs="breadcrumbs"
+          :tabs="tabs"
+          :brief-info="briefInfo"
+          :gallery="gallery"
+          :product="product"
+          :banner="banner"
+          firstColumnText="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged."
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.stories.js]: https://vuejs.github.io/eslint-plugin-vue/rules/attribute-hyphenation.html#rule-details

File: components/04-views/product-view/ProductView.stories.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+          secondColumnText="It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum."
```
Comment (szafran89) [inline components/04-views/product-view/ProductView.stories.js]: https://vuejs.github.io/eslint-plugin-vue/rules/attribute-hyphenation.html#rule-details

File: components/02-elements/swatch/mocks/icon-swatch.json (hunk 1 above)
Comment (Aekal) [reply components/02-elements/swatch/mocks/icon-swatch.json]: In vsf there is `label` with color taken from backend, so I changed `aria-label` to label instead for color and image swatches

File: components/03-modules/additional-content/AdditionalContent.stories.js (hunk 2 above)
Comment (Aekal) [reply components/03-modules/additional-content/AdditionalContent.stories.js]: Done

File: components/04-views/product-view/ProductView.html (hunk 4 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Done

File: components/04-views/product-view/ProductView.html (hunk 5 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Done

File: components/04-views/product-view/ProductView.html (hunk 11 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Done

File: components/04-views/product-view/ProductView.stories.js (hunk 12 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.stories.js]: Done

File: components/04-views/product-view/ProductView.stories.js (hunk 13 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.stories.js]: Done

File: components/03-modules/additional-content/AdditionalContent.vue (hunk 3 above)
Comment (Aekal) [reply components/03-modules/additional-content/AdditionalContent.vue]: Done

File: components/04-views/product-view/ProductView.html (hunk 6 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Should be, I found a bug in heading component validation, fixed

File: components/04-views/product-view/ProductView.html (hunk 7 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Done

File: components/04-views/product-view/ProductView.html (hunk 8 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Done

File: components/04-views/product-view/ProductView.html (hunk 9 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Done

File: components/04-views/product-view/ProductView.html (hunk 10 above)
Comment (Aekal) [reply components/04-views/product-view/ProductView.html]: Done

**Pull request merged**